### PR TITLE
[Feature][Multimodal] CI for running AG Multimodal tests in multi-gpu environment

### PR DESCRIPTION
**Pull request by prateekdesai04**
*Description of changes:*
Adding multi-gpu CI environment for running multimodal tests.
Currently we are running all tests but this might change soon as we plan to remove a few of them to save on instance usage cost.
This PR mainly focuses on the infrastructure changes and also marks some multimodal tests that do not work in multi-gpu environment.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -181,7 +181,7 @@ jobs:
         with:
           job-type: CI-GPU
           job-name: AutoGluon-Multimodal-Predictor
-          command: chmod +x ./.github/workflow_scripts/test_multimodal_predictor.sh && ./.github/workflow_scripts/test_multimodal_predictor.sh
+          command: chmod +x ./.github/workflow_scripts/test_multimodal_predictor.sh && ./.github/workflow_scripts/test_multimodal_predictor.sh --runsgpu
```
Comment (tonyhoo) [inline .github/workflows/continuous_integration.yml:184]: Suggestion: The name "runsgpu" may not be clear in its meaning. It is advisable to consider renaming it to something more obvious, for example, "runonegpu".

**Review by yinweisu — APPROVED** [review]
LGTM on the CI side. I would recommend to ask @zhiqiangdon for a review on the multimodal tests being marked as single gpu.

File: multimodal/tests/unittests/others/test_deployment_torch.py [hunk 2]
```diff
@@ -115,6 +116,7 @@ def test_fusion_model_dump():
     assert os.path.exists(timm_image_dir) and (len(os.listdir(timm_image_dir)) == 2) == True
 
 
+@pytest.mark.sgpu
```
Comment (zhiqiangdon) [inline multimodal/tests/unittests/others/test_deployment_torch.py:119]: `sgpu` is kind of not an intuitive name. How about `single_gpu`?

File: multimodal/tests/unittests/predictor/test_predictor.py [hunk 3]
```diff
@@ -93,6 +93,7 @@ def verify_no_redundant_model_configs(predictor):
     assert sorted(predictor._config.model.names) == sorted(model_names)
 
 
+@pytest.mark.sgpu
```
Comment (zhiqiangdon) [inline multimodal/tests/unittests/predictor/test_predictor.py:96]: We probably need to fix tests like this in the multi-gpu env.

File: .github/workflows/continuous_integration.yml [hunk 4]
```diff
@@ -221,7 +221,67 @@ jobs:
         with:
           job-type: CI-GPU
           job-name: AutoGluon-Multimodal-Others-2
+          command: chmod +x ./.github/workflow_scripts/test_multimodal_others_2.sh && ./.github/workflow_scripts/test_multimodal_others_2.sh --runsgpu
+  test_multimodal_others_mgpu:
```
Comment (zhiqiangdon) [inline .github/workflows/continuous_integration.yml:225]: How about changing `mgpu` to `multi_gpu`, which is more clear?

**Review by zhiqiangdon — COMMENTED** [review]
Have you got the time statistics of running the multi-gpu tests? How do they compare to the single gpu tests?

**Comment by prateekdesai04** [comment]
/benchmark ref=dashboard module=tabular preset=good benchmark=test time_limit=1h

**Comment by prateekdesai04** [comment]
/benchmark ref=automm-ci module=tabular preset=good benchmark=test time_limit=1h

**Comment by prateekdesai04** [comment]
/benchmark ref=master module=tabular preset=good benchmark=test time_limit=1h

**Comment by prateekdesai04** [comment]
/benchmark ref=automm-ci module=tabular preset=good benchmark=test time_limit=1h

**Review by zhiqiangdon — COMMENTED** [review]
Looks like our CI doesn't run multi-gpu tests for this PR, right?

**Review by zhiqiangdon — COMMENTED** [review]
One question: where is the instance type you use in multi-gpu testing? I went over the PR, but can't find the related key word like g4dn.x12large?

**Comment by prateekdesai04** [comment]
> Looks like our CI doesn't run multi-gpu tests for this PR, right?

Once it is merged then it will run the multi-gpu tests. This is a PR to enable this functionality.

**Comment by prateekdesai04** [comment]
> One question: where is the instance type you use in multi-gpu testing? I went over the PR, but can't find the related key word like g4dn.x12large?

This is set up in the AWS Batch side, the job queue CI-MULTI-GPU (in CI/batch/submit-job.py) is linked to a compute environment which is configured to use g4dn.12xlarge.

**Review by zhiqiangdon — APPROVED** [review]
LGTM!

**Pull request merged**